homeassistant/components/obihai/connectivity.py

```diff
@@ -48,11 +48,12 @@ def __init__(
         self.host = host
         self.username = username
         self.password = password
-        self.serial: str
+        self.serial: str = ""
```

Comment:
Why is this initialized here? If we pass this to an entity constructor, it will now silently give itself the wrong unique_id.

Reply:
Thanks, you're right.  I've been seeing some weird errors intermittently and they went away after I added this.  Upon closer investigation I think the errors were related to credential problems and I was covering up the issue on my local system.  I just pushed a commit which removes this default value and should make the credential problem more visible, at least, that's the hope.